### PR TITLE
[NeMo-UX] Use single instance of loss reductions in GPTModel

### PR DESCRIPTION
**Pull request by hemildesai**
# What does this PR do ?

Uses a singleton for `training_loss_reduction` and `validation_loss_reduction` in `GPTModel`.

**Collection**: llm

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [x] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Review by ShriyaPalsamudram — APPROVED** [review]

**Review by ShriyaPalsamudram — DISMISSED** [review]
Functionality works

File: nemo/collections/llm/gpt/model/base.py [hunk 1]
```diff
@@ -25,6 +25,9 @@
 
     from nemo.collections.common.tokenizers.tokenizer_spec import TokenizerSpec
 
+TRAIN_LOSS_REDUCTION = MaskedTokenLossReduction()
+VALIDATION_LOSS_REDUCTION = MaskedTokenLossReduction(validation_step=True)
```
Comment (ericharper) [inline nemo/collections/llm/gpt/model/base.py:29]: Let's comment that this is a temporary WAR.

Comment (ericharper) [reply nemo/collections/llm/gpt/model/base.py:29]: Or remove the globals.

Comment (hemildesai) [reply nemo/collections/llm/gpt/model/base.py]: Updated.

**Review by akoumpa — APPROVED** [review]
LGTM,

Alternatively we could decorate `training_loss_reduction` & `validation_loss_reduction` with `@cache`, but I have no preference. Please proceed with the merge.

Thanks.

**Pull request merged**